File: app/src/main/java/email/schaal/ocreader/view/FeedManageListener.java
package email.schaal.ocreader.view;

import email.schaal.ocreader.database.model.Feed;

/**
 * Callbacks for feed management
 */
public interface FeedManageListener {
    void addNewFeed(String url, long folderId, boolean finishAfterAdd);
    void deleteFeed(Feed feed);
    void changeFeed(long feedId, long folderId);

    void showFeedDialog(Feed feed);
}

File: app/src/main/java/email/schaal/ocreader/view/FeedManageListener.kt
package email.schaal.ocreader.view

import email.schaal.ocreader.database.model.Feed

/**
 * Callbacks for feed management
 */
interface FeedManageListener {
    fun addNewFeed(url: String?, folderId: Long, finishAfterAdd: Boolean)
    fun deleteFeed(feed: Feed?)
    fun changeFeed(feedId: Long, folderId: Long)
    fun showFeedDialog(feed: Feed?)
}